base/show.jl

```diff
@@ -802,10 +802,18 @@ function show_unquoted(io::IO, ex::Expr, indent::Int, prec::Int)
             if (na == 2 || (na > 2 && func in (:+, :++, :*))) &&
                     all(!isa(a, Expr) || a.head !== :... for a in func_args)
                 sep = " $func "
+                # parenthesize (-x) ^ y for negative x
```

Comment:
In addition to custom operators like `↑`, the current behavior is also wrong for other unary ops, not just `-`.  e.g. `:((!a) ^ b)` prints as `:(!a ^ b)`, which has the same problem.

The right thing seems to be to parenthesize `(UNARY x) OP y` for *all* unary operators, but only if `OP` has `operator_precedence(:OP) == operator_precedence(:^)`, because this is [apparently the only kind of operator](https://github.com/JuliaLang/julia/blob/72ecb44f6c24acaab96c8bcc9647fad31000043e/src/julia-parser.scm#L941-L944) that takes precedence over unary operators.